Amplify/Categories/DataStore/Model/Internal/Schema/ModelSchema.swift

```diff
@@ -84,14 +94,16 @@ public struct ModelSchema {
     public let authRules: AuthRules
     public let fields: ModelFields
     public let attributes: [ModelAttribute]
+    public let indexes: [ModelAttribute]
 
     public let sortedFields: [ModelField]
 
-    public var primaryKey: ModelField {
-        guard let primaryKey = fields.first(where: { $1.isPrimaryKey }) else {
+    private var _primaryKey: ModelPrimaryKey?
+    public var primaryKey: ModelPrimaryKey {
```

Comment:
non-model types don't have a primary key, so its value will be `nil`. For those models we don't access the pk value though since we don't store them in separate tables